apollo-federation/src/connectors/json_selection/apply_to.rs

```diff
@@ -954,6 +1010,23 @@ impl ApplyToInternal for WithRange<LitExpr> {
                     literal.compute_output_shape(context, input_shape, dollar_shape.clone());
                 subpath.compute_output_shape(context, literal_shape, dollar_shape)
             }
+
+            LitExpr::OpChain(_, operands) => {
+                // For any coalescing operator chain, the result can be any of the operands
+                let shapes: Vec<Shape> = operands
```

Comment:
Let's go ahead and break this shape code into the different known `LitOp` cases, something like
```rust
LitExpr::OpChain(op, operands) => {
    match op {
        LitOp::NullCoalescing | LitOp::NoneCoalescing => {
            // Same shape code for both of these cases
        }
        // Keep the match exhaustive, with no _ => {} catch-all case
    }
```
Keeping the `match` exhaustive means we will be forced to update it when we add more operators, which is good.